src/lib.rs

```diff
@@ -670,7 +737,29 @@ impl Library {
                         self.include_paths.push(PathBuf::from(inc));
                     }
                 }
-                _ => (),
+                _ => {
+                    let path = std::path::Path::new(part);
+                    if path.is_file() {
+                        // Cargo doesn't have a means to directly specify a file path to link,
+                        // so split up the path into the parent directory and library name.
+                        // TODO: pass file path directly when link-arg library type is stabilized
+                        // https://github.com/rust-lang/rust/issues/99427
+                        if let Some(dir) = path.parent() {
+                            let link_search = format!("rustc-link-search={}", dir.display());
+                            config.print_metadata(&link_search);
```

Comment:
This should probably only be included if the `extract_lib_from_filename()` below succeeds. Otherwise an unused search path is added